app/services/resources/resources.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Services
+  module Resources
+    include Errors
```

Comment:
Rails might auto-load this module, so I think it might be possible to delete this `include Errors` line here. In development mode, Rails lazily loads modules, so when it encounters `raise Errors::PreconditionFailed`, it will go look for the appropriate file and load it into scope.